cpustat.cpp

```diff
@@ -82,10 +82,10 @@ void CpuStatPrivate::updateSources()
         int dash = range.indexOf(QLatin1Char('-'));
         if (dash != -1)
         {
-            uint min = range.leftRef(dash).toUInt(&ok);
+            uint min = range.left(dash).toUInt(&ok);
             if (ok)
             {
-                uint max = range.midRef(dash + 1).toUInt(&ok);
+                uint max = range.mid(dash + 1).toUInt(&ok);
```

Comment:
Qt doc says `QString::sliced` is faster (https://doc.qt.io/qt-6/qstring.html#mid):

```c++
                uint max = range.sliced(dash + 1, range.size() - dash - 1).toUInt(&ok);

```

Reply:
Wjat about `QStringView`?
Ut's te replacement for `QStringRef` after all

Reply:
Not needed here. And there's no trace of `QStringRef` in the code; is there?

Reply:
And if you mean `QString::toInt`, Qt already uses `QStringView` for it. Better to trust Qt doc, instead of enforcing suspicious optimizations ;)

Reply:
> Not needed here. And there's no trace of `QStringRef` in the code; is there?

The original  `midRef()` call returns a `QStringRef`

Reply:
> And if you mean `QString::toInt`, Qt already uses `QStringView` for it. Better to trust Qt doc, instead of enforcing suspicious optimizations ;)

I'm not sure what you mean here.
The problem happens before `toInt()`
It's `mid()` that will copy buffer and build a new `QString`.

I suggest turning `range` in a string view since it doesn't get modified

Reply:
I understand what you say but, as far as I know, there's no up-to-date recommendation about using `QStringView` innstead of `QStringRef`. https://doc.qt.io/qt-6/qtcore-changes-qt6.html is old; it mentions `QStringView::left` and `QStringView::mid`, which are both deprectated.

 On the other hand, the up-to-date Qt doc is explicit about using `QString::first` and `range.sliced`. IMHO, better safe than sorry, by sticking with Qt doc.

Reply:
Replacement was done in qtbase: [qtbase@c49728](https://github.com/qt/qtbase/commit/c49728eb27be0f3f2eaaa77b0ed573f5d8705af1) and [qtbase@690abaac](https://github.com/qt/qtbase/commit/690abaac0e4bd57b5510e68730f7965c01a113da)

Later usage of `mid()` was replaced with `sliced()` in [qtbase@b4d003c](https://github.com/qt/qtbase/commit/b4d003c77f947111b809c9f7c836a221f3981ba2)

Reply:
To me, this difference in opinions is unimportant. As I said, I understand your point but prefer to follow the up-to-date Qt doc.

Fell free to ask @stefonarch to make the changes you prefer (by recommending codes, as I did).